.ci-cd/Dockerfile.cpu

```diff
@@ -31,8 +33,25 @@ RUN pip3 install  pre-commit==1.17.0 \
     pylint==2.3.1 \
     yapf==0.28.0
 
+# Ubuntu 18.04 officially supports git version 2.17 (through apt-get),
+# but github CI checkout@v2 requires a higher version of git (>=2.18).
+# Here we install git 2.18 from source.
+
+# install some pre-requisites for building git from source
+RUN apt-get update && apt-get install -y libssl-dev libghc-zlib-dev \
+    libcurl4-gnutls-dev libexpat1-dev gettext unzip
+
+RUN mkdir -p /opt/local_git \
```

Comment:
Is it possible to directly install using apt-get by adding PPA?

https://itsfoss.com/install-git-ubuntu/

Reply:
> Is it possible to directly install using apt-get by adding PPA?
> 
> https://itsfoss.com/install-git-ubuntu/

Tried this first. Does not work.

Reply:
What I tried was
```
# for add-apt-repository
RUN apt-get install software-properties-common -y
RUN add-apt-repository ppa:git-core/ppa && apt update -y && apt install git
```

Eventually will get an ``apt_pkg`` related error. Therefore I switched to building from source.